src/Tasks/Common/Resources/Strings.resx

```diff
@@ -815,4 +815,8 @@ To install these workloads, run the following command: {1}</value>
     <value>NETSDK1177: Failed to sign apphost with error code {1}: {0}</value>
     <comment>{StrBegin="NETSDK1177: "}</comment>
   </data>
+  <data name="SelfContainedOptionShouldBeUsedWithRuntime" xml:space="preserve">
+    <value>NETSDK1178: The  '--self-contained' or '--no-self-contained' option should be used with '-r|--runtime'.</value>
```

Comment:
@richlander @KathleenDollard To review the warning text here.  Perhaps something like the following would be clearer:

> When the target runtime is specified with '--runtime' or '-r', the deployment type should be specified with '--self-contained' or '--no-self-contained'.

Reply:
@richlander suggested this string in the previous PR reviews.

Reply:
That seems a bit long, and I like to reduce wrapping where possible. Maybe: 

```suggestion
    <value>NETSDK1178: The  '--self-contained' or '--no-self-contained' option should be used when  '--runtime' is specified.</value>
```

Daniel, is there a reason you want to lead with the conditional?

Reply:
@KathleenDollard It sounded a bit better to me, but it could certainly be reversed.

Reply:
I like shorter version better.

How about this one?

> One of '--self-contained' or '--no-self-contained' options are required when '--runtime' is used.

Reply:
I hate to keep this going, but...

> Either '--self-contained' or '--no-self-contained' is required when '--runtime' is used.

Aligning "One of" and "are" grammatically seems hard ;-)